### ISSUE TITLE
How to get native element from a field?

### ISSUE THREAD

**Issue by edbzn** [ngx-formly/ngx-formly#2950]
**Question**

I need to access the native element from a `FormControl` generated with ngx-formly, the concrete use case is the following: I need to scroll to the first field with an error state and focus it in a pretty long form.

A possible solution would be to attach a directive to the `formControl` attribute that assigns the native element:

```ts
@Directive({ selector: '[formControl], [formControlName]' })
export class NativeElementInjectorDirective {
    constructor(private el: ElementRef, private control : NgControl) {
        (control as any).nativeElement = el.nativeElement;
    }
}
```

Any idea how can implement it with ngx-formly?

**Comment by kenisteward** [comment]
On an internal lib we do this in formly by attaching the key name as a
class to every formly-field ( with class name and an extension ) and then
use that key to scroll to the formly field. After that normal tab index
rules take over

I could see simply adding an error class if an error exists on the field
and the just doing a document selector and scrolling to the first one you
find.

**Comment by vitale232** [comment]
Hi @edbzn ,

This doesn't answer your specific question, but it does address your use case. Continuing with your idea of an attribute directive, why not just use the attribute directive to access the native controls? 

Here's an implementation I've used, based off a [blog post by Cory Rylan](https://coryrylan.com/blog/focus-first-invalid-input-with-angular-forms)

```typescript
import { Directive, ElementRef, HostListener } from '@angular/core';

@Directive({
  selector: '[appFindInvalid]'
})
export class FindInvalidDirective {
  private findNodes = ['INPUT', 'SELECT', 'TEXTAREA', 'OPTGROUP'];

  constructor(private el: ElementRef) {}

  @HostListener('submit')
  onFormSubmit(): void {
    const invalidControls: HTMLElement[] = Array.from(
      this.el.nativeElement.querySelectorAll('.ng-invalid')
    );

    if (invalidControls.length > 0) {
      for (const element of invalidControls) {
        if (this.findNodes.includes(element.nodeName)) {
          element.focus();
          return;
        }
      }
    }
}
```

This implementation leverages the fact that [Angular automatically adds the `ng-invalid` CSS class to invalid controls](https://angular.io/api/forms/NgControlStatus). When the form is submitted, it queries the children of the `form` for the `ng-invalid` class, and sets the focus on the first element it finds.

Here it is integrated with [Formly in a Stackblitz](https://stackblitz.com/edit/angular-ob1csv?file=src%2Fapp%2Ffind-invalid.directive.ts). Just scroll to the bottom, hit "Fill with Invalid Content", then click "Submit"... I think this is the effect you're trying to achieve. Note that I declared the `Directive` in the `AppModule` and added it to the form element.
